ci/jenkins/test-vmc.sh

```diff
@@ -185,10 +235,138 @@ function release_static_ip() {
+    setup_aws_sg
+
+    echo "=== namespace value substitution ==="
+    mkdir -p ${GIT_CHECKOUT_DIR}/jenkins/out
+    cp ${GIT_CHECKOUT_DIR}/ci/cluster-api/aws/templates/* ${GIT_CHECKOUT_DIR}/jenkins/out
+    sed -i "s/VPCID/${AWS_VPC_ID}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/SUBNETID/${AWS_SUBNET_ID}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/CONTROLPLANESG/${CONTROLPLANE_SGID}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/NODESG/${NODE_SGID}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+
+    sed -i "s/CLUSTERNAMESPACE/${CLUSTER}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/K8SVERSION/${K8S_VERSION}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/AMIID/${AMI_ID}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/CLUSTERNAME/${CLUSTER}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/SSHAUTHORIZEDKEYS/default/g" ${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml
+    sed -i "s/CLUSTERNAMESPACE/${CLUSTER}/g" ${GIT_CHECKOUT_DIR}/jenkins/out/namespace.yaml
+
+    sleep 15s
+    echo '=== Create a cluster in management cluster ==='
+    kubectl apply -f "${GIT_CHECKOUT_DIR}/jenkins/out/namespace.yaml"
+    kubectl apply -f "${GIT_CHECKOUT_DIR}/jenkins/out/cluster.yaml"
+
+    echo '=== Wait for 20 min to get workload cluster secret ==='
+    for t in {1..20}
+    do
+        sleep 1m
+        echo '=== Get kubeconfig (try for 1m) ==='
+        if kubectl get secret/${CLUSTER}-kubeconfig -n${CLUSTER} ; then
+            kubectl get secret/${CLUSTER}-kubeconfig -n${CLUSTER} -o json \
+            | jq -r .data.value \
+            | base64 --decode \
+            > "${GIT_CHECKOUT_DIR}/jenkins/out/kubeconfig"
+            SECRET_EXIST=true
+            break
+        fi
+    done
+
+    if [[ "$SECRET_EXIST" == false ]]; then
+        echo "=== Failed to get secret ==="
+        saveLogs
+        kubectl delete ns ${CLUSTER}
+        exit 1
+    else
+        export KUBECONFIG="${GIT_CHECKOUT_DIR}/jenkins/out/kubeconfig"
+        echo "=== Waiting for 20 minutes for all nodes to be up ==="
+
+        set +e
+        for t in {1..20}
+        do
+            sleep 1m
+            echo "=== Get node (try for 1m) ==="
+            mdNum="$(kubectl get node --no-headers=true | grep -v master | grep -c none)"
+            if [ "${mdNum}" == "2" ]; then
+                echo "=== Setup workload cluster succeeded ==="
+                CLUSTER_READY=true
+                break
+            fi
+        done
+        set -e
+
+        if [[ "$CLUSTER_READY" == false ]]; then
+            echo "=== Failed to bring up all the nodes ==="
+            saveLogs
+            KUBECONFIG=$KUBECONFIG_PATH kubectl delete ns ${CLUSTER}
+            exit 1
+        fi
+    fi
+}
+
+function setup_vc_cluster() {
     export KUBECONFIG=$KUBECONFIG_PATH
     if [ -z $K8S_VERSION ]; then
-      export K8S_VERSION=v1.25.5
+      export K8S_VERSION=v1.25.1
```

Comment:
Why the default K8s version is downgraded?

Reply:
this was done for testing purpose, corrected it.